src/main/java/it/gov/pagopa/pu/debtpositions/service/DebtPositionServiceImpl.java

```diff
@@ -93,5 +97,89 @@ public PagedDebtPositions getPagedDebtPositionsByIngestionFlowFileId(Long ingest
   public void delete(DebtPosition debtPosition) {
     debtPositionDeleteService.delete(debtPosition);
   }
+
+  @Override
+  public Optional<DebtPosition> getDebtPositionByIupdAndOrganizationId(String iupd, Long organizationId) {
+    return debtPositionRepository.findDebtPositionByIupdOrgAndOrganizationId(iupd, organizationId);
+  }
+
+  @Override
+  public void updateDebtPosition(Long debtPositionId, DebtPositionDTO debtPositionDTO) {
+    DebtPosition debtPosition = debtPositionRepository.findEntityGraphByDebtPositionId(debtPositionId);
+    if (debtPosition == null) {
+      throw new EntityNotFoundException("DebtPosition with id %d not found".formatted(debtPositionId));
+    }
+
+    propagateIdsForDebtPosition(debtPosition, debtPositionDTO);
+
+    debtPositionSaveService.saveDebtPositionDTO(debtPositionDTO);
+  }
+
+  private void propagateIdsForDebtPosition(DebtPosition entity, DebtPositionDTO dto) {
+    if (entity == null || dto == null) return;
+
+    dto.setDebtPositionId(entity.getDebtPositionId());
+
+    Map<Integer, PaymentOption> poByIndex = indexBy(
+      entity.getPaymentOptions(),
+      PaymentOption::getPaymentOptionIndex
+    );
+
+    streamOf(dto.getPaymentOptions())
+      .forEach(poDTO -> {
+        PaymentOption poEntity = poByIndex.get(poDTO.getPaymentOptionIndex());
+        poDTO.setDebtPositionId(dto.getDebtPositionId());
+        if (poEntity != null) {
+          propagateIdsForPaymentOption(poEntity, poDTO);
+        }
+      });
+  }
+
+  private void propagateIdsForPaymentOption(PaymentOption poEntity, PaymentOptionDTO poDTO) {
+    poDTO.setPaymentOptionId(poEntity.getPaymentOptionId());
+
+    Map<String, InstallmentNoPII> instByIud = indexBy(
+      poEntity.getInstallments(),
+      InstallmentNoPII::getIud
+    );
+
+    streamOf(poDTO.getInstallments())
+      .forEach(instDTO -> {
+        InstallmentNoPII instEntity = instByIud.get(instDTO.getIud());
+        instDTO.setPaymentOptionId(poDTO.getPaymentOptionId());
+        if (instEntity != null) {
+          propagateIdsForInstallment(instEntity, instDTO);
+        }
+      });
+  }
+
+  private void propagateIdsForInstallment(InstallmentNoPII instEntity, InstallmentDTO instDTO) {
+    instDTO.setInstallmentId(instEntity.getInstallmentId());
+
+    Map<Integer, Transfer> trByIndex = indexBy(
+      instEntity.getTransfers(),
+      Transfer::getTransferIndex
+    );
+
+    streamOf(instDTO.getTransfers())
+      .forEach(trDTO -> {
+        Transfer trEntity = trByIndex.get(trDTO.getTransferIndex());
```

Comment:
let's fill transfer parent id
```suggestion
        trDTO.setInstallmentId(instDTO.getInstallmentId());
        Transfer trEntity = trByIndex.get(trDTO.getTransferIndex());
```